### DISCUSSION TITLE
[Persistent_cohomology] min_rel_persistence

### DISCUSSION THREAD

**Discussion by mglisse** [GUDHI/gudhi-devel#1209] [Ideas]
We have min_persistence to indicate that we do not care about intervals of length less than some value. However, in a number of cases (including usually the Cech, Rips, etc), the relevant quantity is the relative length of the interval: `death/birth` instead of `death-birth` (it would be the length if we went to log-scale).
We should check if there is a convenient way to let users specify a minimum relative length for intervals. Of course it should not be used when there are negative filtration values. And users can always post-process diagrams to remove the intervals with small relative length or whatever other criterion they like, so that's just if we can do it conveniently.

**Comment by VincentRouvreau** [comment]
_[UPDATE 06/2025] I moved this issue as a discussion._

Should we keep the actual [`min_interval_length` parameter](https://gudhi.inria.fr/doc/latest/class_gudhi_1_1persistent__cohomology_1_1_persistent__cohomology.html#a7867f4a24e12e51d00e54ab56354a288) or not ?
This could also be performed by a sort function ?
Let's see if a discussion can bring some nice solutions.
